scripts/manage_health_checks.py

```diff
@@ -0,0 +1,82 @@
+from itertools import chain
+
+import argparse
+import boto3
+import logging
+import json
+import sys
+
+import more_itertools
+
+from azul import config
+from azul.logging import configure_script_logging
+
+log = logging.getLogger(__name__)
+
+
+class HealthCheckManager:
+
+    def __init__(self):
+        self.route53 = boto3.client('route53')
+
+    def get_health_checks(self):
+        paginator = self.route53.get_paginator('list_health_checks')
+        hcs = chain.from_iterable(page['HealthChecks'] for page in paginator.paginate())
+        batch_size = 10  # Route53 lets us get tags for at most ten resources at a time
+        hc_batches = more_itertools.chunked(hcs, batch_size)
+        dcp_hcs = {}
+        for hc_batch in hc_batches:
+            hc_batch = {hc['Id']: hc for hc in hc_batch}
+            response = self.route53.list_tags_for_resources(ResourceType='healthcheck',
+                                                            ResourceIds=list(hc_batch.keys()))
+            for tag_set in response['ResourceTagSets']:
+                assert tag_set['ResourceType'] == 'healthcheck'
+                for tag in tag_set['Tags']:
+                    if tag['Key'] == 'Name':
+                        hc_name = tag['Value']
+                        hc_id = tag_set['ResourceId']
+                        dcp_hcs[hc_name] = hc_batch[hc_id]
+        return dcp_hcs
+
+    def provision_health_check(self, hcs, dcp_hc, link):
+        dcp_hcs = dcp_hc['HealthCheckConfig']['ChildHealthChecks']
+        azul_hcs = []
+        for hc_name in (config.service_name,
+                        config.indexer_name,
+                        config.data_browser_name,
+                        config.data_portal_name):
+            azul_hcs.append(hcs[hc_name]['Id'])
+
+        dcp_hcs = list(set(dcp_hcs).difference(set(azul_hcs)) if not link else set(dcp_hcs).union(set(azul_hcs)))
```

Comment:
See if you can simplify this by making the variables sets instead of converting to sets. 

Also see if you can use operators instead of methods to perform set operations:

```
{1,2}-{2,3}
{1}
```